crypto/evp/pmeth_gn.c

```diff
@@ -18,36 +21,144 @@
 #include "evp_local.h"
 
 #ifndef FIPS_MODE
```

Comment:
Maybe this should be moved to just surround the legacy code (or this another PR?)

Reply:
Actually, we don't even really need to do that.  Have a look t, oh I dunno, `crypto/evp/signature.c`...  there's unguarded legacy code there, the only "guard" is the check to see if `cts->pmeth` is NULL...

We *might* want to have a closer look at that, 'cause there's really no point having legacy code go through validation...  but in the mean time, I'll take away these guards.

Reply:
...  and the more I think about it, the more I think we *should* guard legacy code with this check, so I'll start with this function.  Adjustments that aren't directly part of this PR will have to be done in another.

Reply:
yep I will do it now..